src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -89,16 +93,24 @@ private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progre
                     StartedOrStopped(started: true);
                     break;
                 case ProgressStatus.PendingItemCountUpdated:
-                    _lastPendingItemCount = progressData.PendingItemCount.Value;
-                    ProgressUpdated();
+                    lock (_progressUpdatedLock)
+                    {
+                        _lastPendingItemCount = progressData.PendingItemCount.Value;
+                        ProgressUpdated();
```

Comment:
This is a call to external code from inside a lock. I'm not sure we want that.

Reply:
Is the concern that we're calling to the editor to report the task center status under the lock?  Would it be more appropriate to more finely scope the lock to enqueue refresh / where the delay gets changed for now, then followup with a rework for 16.5?